tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -297,6 +297,33 @@ def test_subscription(self):
             project=project,
         ).value == 0
 
+    def test_invalid_platform(self):
+        """
+        Ensure platform validation works
+        """
+        project = self.project  # force creation
+        self.login_as(user=self.user)
+        url = reverse(
+            'sentry-api-0-project-details',
+            kwargs={
+                'organization_slug': project.organization.slug,
+                'project_slug': project.slug,
+            }
+        )
+        platform_name = 'notvalid'
+        resp = self.client.put(
+            url, data={
+                'name': 'hello world',
+                'slug': 'foobar',
+                'platform': platform_name,
```

Comment:
I don't know how to ensure an error is given to raven. Does anyone know of an example I can look at? 
@denamwangi